src/HttpMethodsClient.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Http\Client;
+
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\UriInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Interface for HTTP conventional methods
+ *
+ * Use this interface when you do not have PSR-7 RequestInterface instances available.
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface HttpMethodsClient
+{
+    /**
+     * Sends a GET request
+     *
+     * @param string|UriInterface $uri
+     * @param array               $headers
+     * @param array               $options
+     *
+     * @throws Exception
+     *
+     * @return ResponseInterface
+     */
+    public function get($uri, array $headers = [], array $options = []);
```

Comment:
HEAD is just a different kind of GET. but yeah, lets not cover these
cases here.